app/controllers/reports_controller.rb

```diff
@@ -0,0 +1,21 @@
+class ReportsController < ApplicationController
+
+  def index;
+    @categories = Category.order(:id).all
+  end
+
+  def create;
+    set_parameters[:report]
+    report = CustomersWithoutOrders.new(
+        start_date: @parameters[:start_date],
+        end_date: @parameters[:end_date],
+        category: @parameters[:category])
+    @customers = report.call
+  end
+
+  private
+
+  def set_parameters
```

Comment:
Generalnie cały myk z tym polega na działaniu `||=`:
```
irb(main):001:0> @whatever
=> nil
irb(main):002:0> @whatever ||= "whatever"
=> "whatever"
irb(main):003:0> @whatever ||= "something different"
=> "whatever"
```
Generalnie używa się tego w taki sposób:
```
private

def parameters
  @parameters ||= ...
end
```
I teraz jak wywołujesz metodę `parameters` to jeśli jest to pierwszy raz to @parameters się inicjalizuje na odpowiednią wartość (bo jest nilem), a za każdym kolejnym razem ta zmienna jest już zainicjalizowana, więc zwracana jest tylko jej wartość.
Poza tym masz keyword arguments w `CustomersWithoutOrders`. Więc może to wyglądać następująco:
```
def create
  @customers = CustomersWithoutOrders.new(parameters).call
end
```

Reply:
Done

Reply:
No chyba jednak nie? :smile: